addon/components/o-s-s/illustration.ts

```diff
@@ -0,0 +1,36 @@
+import Component from '@glimmer/component';
+
+const extractCSSVars = (): string[] => {
+  return (
+    Array.from(document.styleSheets)
+      .filter((styleSheet) => {
+        try {
+          return styleSheet.cssRules;
+        } catch (e) {
+          return false;
+        }
+      })
+      .map((styleSheet) => Array.from(styleSheet.cssRules))
+      .flat()
+      .filter((cssRule: CSSStyleRule) => cssRule.selectorText === ':root')
+      .map((cssRule: CSSStyleRule) => cssRule.cssText.split('{')[1].split('}')[0].trim().split(';'))
+      .flat()
+      .filter((text) => text !== '')
+  );
+};
+
+interface OSSIllustrationArgs {
+  src: string;
+}
+
+export default class OSSIllustration extends Component<OSSIllustrationArgs> {
+  setupCSSVars(event: Event): void {
+    const svgDocument = (<HTMLObjectElement>event.target).contentDocument?.querySelector('svg');
+
+    if (svgDocument) {
+      const style = document.createElement('style');
```

Comment:
So, for every `<OSS::Illustration />` used, we're going to inject another style tag ? 🤔 
Why do we do that ? Can't the `<object>` tag already access the css vars that are already in the page ? 🤔